src/gui/parameters_screen.py

```diff
@@ -326,28 +418,257 @@ def validate_point_side_size(self):
+
+        def validate_longitude_oil_source(self):
+            value = self.longitude_oil_source_input.get()
+            if not value:
+                return False
+            if -180 <= float(value) <= 180:
+                self.longitude_oil_source = float(value)
+                self.longitude_oil_source_validation_label.config(text="Valid value", fg="black")
+                self.correctly_set_parameters_oil_sources[0] = 1
+                self.check_all_parameters_validity_oil_sources()
+                return True
+            self.longitude_oil_source_validation_label.config(text="Invalid value", fg="red")
+            self.oil_sources_listbox_insert.config(state=DISABLED)
+            self.correctly_set_parameters_oil_sources[0] = 0
+
+        def validate_latitude_oil_source(self):
+            value = self.latitude_oil_source_input.get()
+            if not value:
+                return False
+            if -90 <= float(value) <= 90:
+                self.latitude_oil_source = float(value)
+                self.latitude_oil_source_validation_label.config(text="Valid value", fg="black")
+                self.correctly_set_parameters_oil_sources[1] = 1
+                self.check_all_parameters_validity_oil_sources()
+                return True
+            self.latitude_oil_source_validation_label.config(text="Invalid value", fg="red")
+            self.oil_sources_listbox_insert.config(state=DISABLED)
+            self.correctly_set_parameters_oil_sources[1] = 0
+
+        def validate_mass_per_minute_oil_source(self):
+            value = self.mass_per_minute_oil_source_input.get()
+            if not value:
+                return False
+            if float(value) > 0:
+                self.mass_per_minute_oil_source = float(value)
+                self.mass_per_minute_oil_source_validation_label.config(text="Valid value", fg="black")
+                self.correctly_set_parameters_oil_sources[2] = 1
+                self.check_all_parameters_validity_oil_sources()
+                return True
+            self.mass_per_minute_oil_source_validation_label.config(text="Invalid value", fg="red")
+            self.oil_sources_listbox_insert.config(state=DISABLED)
+            self.correctly_set_parameters_oil_sources[2] = 0
+
+        def validate_spill_start_oil_source(self):
+            value = self.spill_start_oil_source_input.get()
+            if not value:
+                return False
+            if re.match('\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}', value):
+                try:
+                    pd.Timestamp(value)
+                    self.spill_start_oil_source = value
+                    self.spill_start_oil_source_validation_label.config(text="Valid value", fg="black")
+                    self.correctly_set_parameters_oil_sources[3] = 1
+                    self.check_all_parameters_validity_oil_sources()
+                    return True
+                except ValueError:
+                    pass
```

Comment:
shouldn't that return False instead of passing?

Reply:
It should, but it doesn't matter because this information is not used anywhere due to different approach to validation